pytorch_lightning/callbacks/quantization.py

```diff
@@ -318,3 +326,82 @@ def on_predict_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule
         if "predict" in self._observer_disabled_stages:
             self._restore_last_observer_enabled()
+
+
+class INCQuantization(Callback):
+    """Quantization allows speeding up inference and decreasing memory requirements by performing computations and
+    storing tensors at lower bitwidths (such as INT8) than floating point precision. And this callback will
+    quantized model with Intel Neural Compressor(INC) tool.
+
+    Args:
+
+        config_path_or_obj: config file or Quantization_Conf class for INC:
+
+        metric_name: Specified metric which will be computed in evaluation function.
+
+        save_path: save path where the quantization config and weights be saved to
+
+    .. Intel Neural Compressor: https://github.com/intel/neural-compressor
+    """
+
+    def __init__(
+        self,
+        config_path_or_obj,
+        metric_name: str,
+        save_path: str = None,
+    ) -> None:
+        try:
+            from neural_compressor.conf.config import Quantization_Conf
+        except ImportError as e:
+            raise RuntimeError("Please install neural-compressor with: pip install neural-compressor") from e
+
+        self.config = (
+            config_path_or_obj
+            if isinstance(config_path_or_obj, Quantization_Conf)
+            else Quantization_Conf(config_path_or_obj)
+        )
+        self.approach = self.config.usr_cfg.quantization.approach
+        assert metric_name is not None, "Please set metric name for evaluation loop!"
+        self.metric_name = metric_name
+        self.save_path = save_path
+
+    def on_compress_model(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
+        """Called when the compressor loop begins."""
+        try:
+            from neural_compressor.experimental import common, Quantization
+        except ImportError as e:
+            raise RuntimeError("Please install neural-compressor with: pip install neural-compressor") from e
+
+        def eval_func(model):
+            pl_module_tmp = copy.deepcopy(pl_module)
+            pl_module_tmp.model = model
+            out = trainer.validate(pl_module_tmp)
+            trainer.state.fn = TrainerFn.COMPRESSING
+            return out[0][self.metric_name]
+
+        def train_func(model):
+            pl_module_tmp = copy.deepcopy(pl_module)
+            pl_module_tmp.model = model
+            trainer.fit(pl_module_tmp)
+            trainer.state.fn = TrainerFn.COMPRESSING
+            return pl_module_tmp.model
+
+        assert hasattr(
+            pl_module, "model"
+        ), "The LightningModule should have the backbone model which is a native PyTorch model!"
+        quantizer = Quantization(self.config)
+        quantizer.model = common.Model(pl_module.model)
+
+        if self.approach == QuantizationMode.PTQ_STATIC.value:
+            quantizer.calib_dataloader = pl_module.train_dataloader()
```

Comment:
This doesn't seem very pythonic. Any better API to setup the quantizer ?

Reply:
I changed quantizer.model = common.Model(pl_module.model) to quantizer.model = getattr(pl_module.model)